### PR TITLE
[no squash] Random things, mostly fixes

### PR DESCRIPTION
**Pull request by sfan5**
sorry for the wild collection but I'm not going to create fifteen PRs.

## To do

This PR is Ready for Review.

### CODE REVIEW

File: src/script/lua_api/l_settings.cpp [hunk 1]
```diff
@@ -54,6 +56,17 @@ static inline int checkSettingSecurity(lua_State* L, const std::string &name)
 		return -1;
 	}
 
+	const char *disallowed[] = {
+		"main_menu_script", "shader_path", "texture_path", "screenshot_path",
+		"serverlist_file", "serverlist_url", "map-dir", "contentdb_url",
```
Comment (appgurueu) [inline src/script/lua_api/l_settings.cpp:61]: Is the dash in `map-dir` a typo? (also what about using a hash map instead?)

Comment (sfan5) [reply src/script/lua_api/l_settings.cpp:61]: > Is the dash in map-dir a typo?

no https://github.com/minetest/minetest/blob/8b74257bf3cbb54e78614ac6aaaba79adf75cc8e/src/main.cpp#L731

> what about using an unordered set instead?

meh

File: src/server/luaentity_sao.cpp [hunk 2]
```diff
@@ -402,6 +392,20 @@ std::string LuaEntitySAO::getDescription()
 void LuaEntitySAO::setHP(s32 hp, const PlayerHPChangeReason &reason)
 {
 	m_hp = rangelim(hp, 0, U16_MAX);
+
+	sendPunchCommand();
```
Comment (SmallJoker) [inline src/server/luaentity_sao.cpp:396]: This might be incorrect. HP changes are not always punches - especially when the HP increases.

Comment (sfan5) [reply src/server/luaentity_sao.cpp:396]: https://github.com/minetest/minetest/blob/ef22c0206f225dbccd67bff9fb888867c63039b3/src/client/content_cao.cpp#L1849-L1885
Looks like the client code handles this correctly.

File: src/server.cpp [hunk 3]
```diff
@@ -2290,31 +2288,33 @@ void Server::sendAddNode(v3s16 p, MapNode n, std::unordered_set<u16> *far_player
...
 			NodeMetadata *meta = m_env->getMap().getNodeMetadata(pos);
 
 			if (!meta)
 				continue;
 
 			v3s16 block_pos = getNodeBlockPos(pos);
-			if (!client->isBlockSent(block_pos) || (player &&
-					player_pos.getDistanceFrom(intToFloat(pos, BS)) > maxd)) {
+			if (!client->isBlockSent(block_pos) ||
+					player_pos.getDistanceFrom(pos) > far_d_nodes) {
```
[the remark this answers is not in the document]
Comment (sfan5) [reply src/server.cpp:2317]: If there is no SAO then `!client->isBlockSent()` should always be true anyway, but I can readd the check

File: src/server/luaentity_sao.cpp (hunk 2 above)
Comment (nerzhul) [reply src/server/luaentity_sao.cpp:396]: i agree with @SmallJoker don't trust the client

Comment (SmallJoker) [reply src/server/luaentity_sao.cpp:396]: @nerzhul This is not about trust.
---

I see. The function simply got an unfortunate naming. It's just about sending the HP update packet; nothing else. A better name for `AO_CMD_PUNCHED` would be `AO_CMD_SET_HP`.

**Comment by SmallJoker** [comment]
"Immediately activate blocks when a player joins"

This commit does not result in any noticeable change. Neither in singleplayer nor when joining an already started server. It always takes about 2 seconds from block render to entity loading. (using `dedicated_server_step = 0.08`)

**Comment by sfan5** [comment]
> his commit does not result in any noticeable change.

Yes. I tested it too and it seems to be quite random, probably dependent on disk speed.
In any case without this commit the world is _never_ loaded on time, while with it there is at least a possibility. Remembering the blocks to be activated is more complicated so I opted not to implement this.

**Comment by SmallJoker** [comment]
In that case, might it make sense to add a `TODO` comment to indicate partially the only working feature? That way it's not forgotten in the future to improve the reliability.

**Comment by sfan5** [comment]
Sure, done.

**Review by SmallJoker — APPROVED** [review]
Respawn works
Node metadata + updates work
Detaching + `on_death` callback works

**Comment by runsy** [comment]
sfan look this error:

[https://github.com/runsy/petz/issues/106#issuecomment-1154866867](https://github.com/runsy/petz/issues/106#issuecomment-1154866867)

The bug was when I do:

self.object:set_hp(0) in an entity of already 0 or less hp.

This was not sense, I know,  but before this PR not crash, and it should not be.

**Comment by runsy** [comment]
I thing  is caused by this line:

markForRemoval();

The entity is removed when set_hp but before this pR not. It was mandatory to use: self.object:remove()

**Comment by sfan5** [comment]
`get_hp()` should always return 0 in the death callback. However if you call `set_hp(0)` that immediately invalidates the object, this is indeed a behavioral change.
I'll verify the former but other than that everything is in order.

**Comment by runsy** [comment]
> `get_hp()` should always return 0 in the death callback. However if you call `set_hp(0)` that immediately invalidates the object, this is indeed a behavioral change. I'll verify the former but other than that everything is in order.

I would just like to emphasize that this was not the case before, at least not in that step.

This can give failures in entity mods from 5.6 onwards.

**Pull request merged**